packages/use-dataloader/src/__tests__/useDataLoader.test.tsx

```diff
@@ -60,25 +58,19 @@ describe('useDataLoader', () => {
     expect(result.current.isLoading).toBe(false)
   })
 
-  test('should render correctly without valid key', async () => {
-    const { result, waitForNextUpdate } = renderHook<
-      UseDataLoaderHookProps,
-      UseDataLoaderResult
-    >(props => useDataLoader(props.key, props.method), {
-      initialProps: {
-        ...initialProps,
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-expect-error
-        key: 2,
+  test('should render correctly without valid key', () => {
+    const { result } = renderHook<UseDataLoaderHookProps, UseDataLoaderResult>(
+      props => useDataLoader(props.key, props.method),
+      {
+        initialProps: {
+          ...initialProps,
+          // @ts-expect-error
+          key: 2,
+        },
+        wrapper,
       },
-      wrapper,
-    })
-    expect(result.current.data).toBe(undefined)
-    expect(result.current.isLoading).toBe(true)
-    await waitForNextUpdate()
-    expect(result.current.data).toBe(undefined)
-    expect(result.current.isSuccess).toBe(true)
-    expect(result.current.isLoading).toBe(false)
+    )
+    expect(result.error?.message).toBe('Key should be a string')
```

Comment:
Here you can probably import `KEY_IS_NOT_STRING_ERROR` constant and check trough it